### PR TITLE
Support to mysql databases

### PR DESCRIPTION
**Pull request by cphyc**
This brings back support to non-sqlite databases.
Notably, this allows to run the database on one server with the data on another.

TODO:
- [x] rename environment variables to `TANGOS_`

### CODE REVIEW

File: tangos/core/creator.py [hunk 1]
```diff
@@ -10,11 +10,11 @@ class Creator(Base):
     __tablename__ = 'creators'
 
     id = Column(Integer, primary_key=True)
-    command_line = Column(String)
+    command_line = Column(String(128))
```
Comment (cphyc) [inline tangos/core/creator.py:13]: I am not sure about this, in particular the `command_line` may be longer than 128 character. Should it be a `Text` instead?

Comment (apontzen) [reply tangos/core/creator.py:13]: Yes I suppose it should be `Text`. In `sqlite` I think `TEXT` and `VARCHAR` are identical so this change should not cause any problems

File: tangos/core/creator.py [hunk 2]
```diff
@@ -10,11 +10,11 @@ class Creator(Base):
...
     dtime = Column(DateTime)
-    host = Column(String)
-    username = Column(String)
-    cwd = Column(String)
+    host = Column(String(128))
+    username = Column(String(128))
+    cwd = Column(String(128))
```
Comment (apontzen) [inline tangos/core/creator.py:17]: This perhaps should also be `Text`

File: tangos/core/simulation.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ class Simulation(Base):
     # __table_args__ = {'useexisting': True}
 
     id = Column(Integer, primary_key=True)
-    basename = Column(String)
+    basename = Column(String(128))
```
Comment (apontzen) [inline tangos/core/simulation.py:21]: Text here too

File: tangos/core/simulation.py [hunk 4]
```diff
@@ -111,7 +111,7 @@ class SimulationProperty(Base):
     data_float = Column(Float)
     data_int = Column(Integer)
     data_time = Column(DateTime)
-    data_string = Column(String)
+    data_string = Column(String(128))
```
Comment (apontzen) [inline tangos/core/simulation.py:114]: Text here too

File: tangos/core/timestep.py [hunk 5]
```diff
@@ -14,7 +14,7 @@ class TimeStep(Base):
     __tablename__ = 'timesteps'
 
     id = Column(Integer, primary_key=True)
-    extension = Column(String)
+    extension = Column(String(128))
```
Comment (apontzen) [inline tangos/core/timestep.py:17]: Text here too

File: tangos/core/creator.py (hunk 2 above)
Comment (cphyc) [reply tangos/core/creator.py]: Are you suggesting to turn `host` and `username` to `Text` fields as well?

File: tangos/testing/__init__.py [hunk 6]
```diff
@@ -160,12 +162,21 @@ def init_blank_db_for_testing(**init_kwargs):
...
+        except OSError:
+            pass
 
+        core.init_db(f"sqlite:///{db_name}", **init_kwargs)
+    else:
+        db_url = f"{testing_db_backend}://{testing_db_user}:{testing_db_password}@localhost"
+        engine = create_engine(db_url)
+        with engine.connect() as conn:
+            conn.execute("COMMIT")
+            # Do not substitute user-supplied database names here.
```
Comment (apontzen) [inline tangos/testing/__init__.py:178]: I'm not sure I understand this comment, what's it saying?

File: tests/test_live_calculation_link_syntax.py [hunk 7]
```diff
@@ -103,6 +106,7 @@ def test_link_returned_halo_is_usable():
 
 def test_multi_calculation_link_returned_halo_is_usable():
     all_links = db.get_timestep("sim/ts1").calculate_all('link(testlink)')
+    print("returned -->",all_links[0][0])
```
Comment (apontzen) [inline tests/test_live_calculation_link_syntax.py:109]: print statement to be removed

**Comment by apontzen** [comment]
I fixed a performance issue and have left two ultra-minor comments above - as soon as they are done I think this is good to merge. Hooray!

File: tangos/testing/__init__.py (hunk 6 above)
Comment (cphyc) [reply tangos/testing/__init__.py:178]: I think this is a leftover from a previous version, I'm going to delete it.

**Pull request merged**